Source/Core/Core/PowerPC/MMU.cpp

```diff
@@ -938,36 +938,23 @@ bool MMU::IsOptimizableRAMAddress(const u32 address, const u32 access_size) cons
 }
 
 template <XCheckTLBFlag flag>
-bool MMU::IsRAMAddress(u32 address, bool translate)
+bool MMU::IsEffectiveRAMAddress(u32 address)
 {
-  if (translate)
-  {
-    auto translate_address = TranslateAddress<flag>(address);
-    if (!translate_address.Success())
-      return false;
-    address = translate_address.address;
-  }
+  auto translate_address = TranslateAddress<flag>(address);
+  if (!translate_address.Success())
+    return false;
+  return IsPhysicalRAMAddress(translate_address.address);
+}
 
-  u32 segment = address >> 28;
-  if (m_memory.GetRAM() && segment == 0x0 && (address & 0x0FFFFFFF) < m_memory.GetRamSizeReal())
-  {
-    return true;
-  }
-  else if (m_memory.GetEXRAM() && segment == 0x1 &&
-           (address & 0x0FFFFFFF) < m_memory.GetExRamSizeReal())
-  {
-    return true;
-  }
-  else if (m_memory.GetFakeVMEM() && ((address & 0xFE000000) == 0x7E000000))
-  {
-    return true;
-  }
-  else if (m_memory.GetL1Cache() && segment == 0xE &&
-           (address < (0xE0000000 + m_memory.GetL1CacheSize())))
-  {
-    return true;
-  }
-  return false;
+bool MMU::IsPhysicalRAMAddress(const u32 address) const
+{
+  const u32 segment = address >> 28;
+  const u32 i = address & 0x0FFFFFFF;
+  return (m_memory.GetRAM() && segment == 0x0 && i < m_memory.GetRamSizeReal()) ||
+         (m_memory.GetEXRAM() && segment == 0x1 && i < m_memory.GetExRamSizeReal()) ||
+         (m_memory.GetFakeVMEM() && (address & 0xFE000000) == 0x7E000000) ||
+         (m_memory.GetL1Cache() && segment == 0xE &&
+          address < 0xE0000000 + m_memory.GetL1CacheSize());
```

Comment:
This eleven-condition-long statement is very dense, I feel it read better as a series of early returns. Also, computing the result of `address & 0x0FFFFFFF` before any conditions is *technically* a tiny bit less efficient for the FakeVMEM and L1Cache code paths.

Reply:
Computing an extra AND here shouldn't actually cost anything. It can run in parallel with some other instruction. It doesn't cost anything in code size either - on the contrary, doing the AND before branching should reduce code size.

I do agree that the code is very dense to read, though.

Reply:
What would be good names for the three scenarios where there used to be early returns? Also, what does `i` represent?

Reply:
> What would be good names for the three scenarios where there used to be early returns?

Why would they need names?

> Also, what does `i` represent?

The masked address. (The top 4 bits aren't hooked up to hardware address lines.)

Reply:
> Why would they need names?
I was going to turn each of them into variables to make it more readable

Reply:
That feels like a strange way to go for code of this kind.

Reply:
Alright, I can comprimise. What would be fitting comments to put above each of the early returns?

Reply:
Sorry, but I don't get it. Why should they have comments? It just feels redundant. And neither the current version of this PR nor mitaclaw's suggestion of moving back to a series of early returns involves having comments, so I don't see how it's a compromise either.

Reply:
I don't see a reason to use the early returns since I don't understand the conditions. If I can put explaining labels in the form of either variable names or comments I'd be happier with it.

Reply:
These conditions are pretty basic if you understand how memory lookups work. While it's usually good to make things clearer for people who don't have much knowledge in the area yet, there is a limit to how far it should be taken. We expect people who edit the MMU.cpp file to have a basic understanding of what an MMU does, because if we can't expect that, we would be needing to do extreme amounts of explaining.